bazel/envoy_build_system.bzl

```diff
@@ -0,0 +1,39 @@
+# References to Envoy external dependencies should be wrapped with this function.
+def envoy_external_dep_path(dep):
+    return "//external:%s" % dep
```

Comment:
How does this help in the use case here, where we need the same name for a given target, one in which we get it from a git_new_repository, and another that is a cc_library hack that pulls in the prebuilt .a files?